inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ManagerFetcher.java

```diff
@@ -0,0 +1,223 @@
+                    DEFAULT_AGENT_MANAGER_VIP_HTTP_PREFIX_PATH);
+            int timeout = conf.getInt(AGENT_MANAGER_REQUEST_TIMEOUT,
+                    DEFAULT_AGENT_MANAGER_REQUEST_TIMEOUT);
+            String secretId = conf.get(MANAGER_AUTH_SECRET_ID);
+            String secretKey = conf.get(MANAGER_AUTH_SECRET_KEY);
+            httpManager = new HttpManager(managerAddr, managerHttpPrefixPath, timeout, secretId, secretKey);
+            baseManagerUrl = httpManager.getBaseUrl();
+            staticConfigUrl = buildStaticConfigUrl(baseManagerUrl);
+            clusterName = conf.get(CLUSTER_NAME);
+        } else {
+            throw new RuntimeException("init manager error, cannot find required key");
+        }
+    }
+
+    private boolean requiredKeys(InstallerConfiguration conf) {
+        return conf.hasKey(MANAGER_ADDR);
+    }
+
+    /**
+     * build config url for manager according to config
+     *
+     * example - http://127.0.0.1:8080/inlong/manager/openapi/installer/getConfig
+     */
+    private String buildStaticConfigUrl(String baseUrl) {
+        return baseUrl + conf.get(INSTALLER_MANAGER_CONFIG_HTTP_PATH, DEFAULT_INSTALLER_MANAGER_CONFIG_HTTP_PATH);
+    }
+
+    /**
+     * request manager to get commands, make sure it is not throwing exceptions
+     */
+    public ConfigResult getConfig() {
+        LOGGER.info("getConfig start");
+        String resultStr = httpManager.doSentPost(staticConfigUrl, getFetchRequest(null));
+        LOGGER.info("ConfigUrl {}", staticConfigUrl);
+        JsonObject resultData = getResultData(resultStr);
+        JsonElement dataElement = resultData.get(AGENT_MANAGER_RETURN_PARAM_DATA);
+        LOGGER.info("getConfig end");
+        if (dataElement != null) {
+            LOGGER.info("getConfig not null {}", resultData);
+            return GSON.fromJson(dataElement.getAsJsonObject(), ConfigResult.class);
+        } else {
+            LOGGER.error("getConfig result does not contain the data field ");
+            return null;
+        }
+    }
+
+    /**
+     * form file command fetch request
+     */
+    public TaskRequest getFetchRequest(List<CommandEntity> unackedCommands) {
+        TaskRequest request = new TaskRequest();
+        request.setAgentIp(localIp);
+        request.setUuid(uuid);
+        request.setClusterName(clusterName);
+        request.setPullJobType(PullJobTypeEnum.NEW.getType());
+        request.setCommandInfo(unackedCommands);
+        return request;
+    }
+
+    /**
+     * thread for config fetcher.
+     *
+     * @return runnable profile.
+     */
+    private Runnable configFetchThread() {
+        return () -> {
+            Thread.currentThread().setName("ManagerFetcher");
+            while (isRunnable()) {
+                try {
+                    int configSleepTime = conf.getInt(AGENT_FETCHER_INTERVAL, DEFAULT_AGENT_FETCHER_INTERVAL);
+                    ConfigResult config = getTestConfig();
```

Comment:
```suggestion
            int configSleepTime = conf.getInt(AGENT_FETCHER_INTERVAL, DEFAULT_AGENT_FETCHER_INTERVAL);
                    ConfigResult config = getTestConfig();
            while (isRunnable()) {
                try {
                    
```

Reply:
This configuration needs to be dynamically updated